### PR TITLE
DTSPO-161 - Adding ability to set key vault name and resource group.

### PR DESCRIPTION
**Pull request by msl8r**
### JIRA link ###
https://tools.hmcts.net/jira/browse/DTSPO-161

### Change description ###
Adding ability to set key vault name and resource group. 
Tested that the module uses cnp-core-infra or core-infra-prod when the key vault name and resource group variables are not set. 
https://dev.azure.com/hmcts/DevOps/_build/results?buildId=98176&view=logs&j=7d39b002-b934-56bd-191d-77d27c532b36&t=0a6fd415-1d51-51d3-6b01-454e8f30270e

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -4,12 +4,14 @@ locals {
 
   db_rules = "${null_resource.subnet_mappings.*.triggers}"
 
-  vaultName = "infra-vault-${var.subscription}"
+  vaultName = var.key_vault_name != null ? var.key_vault_name : "infra-vault-${var.subscription}"
```
Comment (timja) [inline main.tf:7]: don't you want to check empty string ""? or does that get converted to null

File: main.tf [hunk 2]
```diff
@@ -4,12 +4,14 @@ locals {
...
 }
 
 data "azurerm_key_vault" "infra_vault" {
-  name                = "${local.vaultName}"
-  resource_group_name = "${var.env == "prod" || var.env == "idam-prod" || var.env == "idam-prod2" ? "core-infra-prod" : "cnp-core-infra"}"
+  name = "${local.vaultName}"
+  resource_group_name = var.key_vault_rg != null ? var.key_vault_rg : (
```
Comment (timja) [inline main.tf:12]: can you convert this to a local variable please so it sits next to the vault name

File: main.tf (hunk 1 above)
Comment (msl8r) [reply main.tf:7]: Nope you are right I need "" as I've set the default to ""

File: main.tf [hunk 3]
```diff
@@ -4,12 +4,15 @@ locals {
 
   db_rules = "${null_resource.subnet_mappings.*.triggers}"
 
-  vaultName = "infra-vault-${var.subscription}"
+  vaultName = var.key_vault_name != "" ? var.key_vault_name : "infra-vault-${var.subscription}"
+  resource_group_name = var.key_vault_rg != "" ? var.key_vault_rg : (
```
Comment (timja) [inline main.tf:8]: should specify that this is for vault
```suggestion
  vault_resource_group_name = var.key_vault_rg != "" ? var.key_vault_rg : (
```

**Review by timja — APPROVED** [review]

**Pull request merged**